fetch.bs

```diff
@@ -175,8 +175,55 @@ lt="authentication entry">authentication entries</a> (for HTTP authentication).
 
  <dt><dfn for="fetch params">task destination</dfn> (default null)
  <dd>Null, a <a for=/>global object</a>, or a <a for=/>parallel queue</a>.
+ <dt><dfn for="fetch params">timing info</dfn>
+ <dd>A <a for=/>fetch timing info</a>.
 </dl>
 
+<p>A <dfn export>fetch timing info</dfn> is a <a for=/>struct</a> used to maintain timing information
+later required by the resource timing and navigation timing specs. It has the following <a for=struct>items</a>:
+<dl>
+  <dt><dfn for="fetch timing info">redirect start time</dfn> (default zero)
+  <dd>A <a for=/>DOMHighResTimeStamp</a>
+  <dt><dfn for="fetch timing info">redirect end time</dfn> (default zero)
+  <dd>A <a for=/>DOMHighResTimeStamp</a>
+  <dt><dfn for="fetch timing info">fetch start time</dfn> (default zero)
+  <dd>A <a for=/>DOMHighResTimeStamp</a>
+  <dt><dfn for="fetch timing info">request start time</dfn> (default zero)
+  <dd>A <a for=/>DOMHighResTimeStamp</a>
+  <dt><dfn for="fetch timing info">response start time</dfn> (default zero)
+  <dd>A <a for=/>DOMHighResTimeStamp</a>
+  <dt><dfn for="fetch timing info">response end time</dfn> (default zero)
+  <dd>A <a for=/>DOMHighResTimeStamp</a>
+  <dt><dfn for="fetch timing info">worker start time</dfn> (default zero)
+  <dd>A <a for=/>DOMHighResTimeStamp</a>
+  <dt><dfn for="fetch timing info">encoded body size</dfn> (default zero)
+  <dd>A number.
+  <dt><dfn for="fetch timing info">decoded body size</dfn> (default zero)
+  <dd>A number.
+  <dt><dfn for="fetch timing info">connection timing info</dfn> (default null)
+  <dd>A <a for=/>connection timing info</a>.
+</dl>
+
+<p>A <dfn>connection timing info</dfn> is a <a for=/>struct</a> used to maintain timing information
+pertaining to the process of obtaining a connection. It has the following <a for=struct>items</a>:
+<dl>
+  <dt><dfn for="connection timing info">domain lookup start time</dfn> (default zero)
+  <dd>A <a for=/>DOMHighResTimeStamp</a>
+  <dt><dfn for="connection timing info">domain lookup end time</dfn> (default zero)
+  <dd>A <a for=/>DOMHighResTimeStamp</a>
+  <dt><dfn for="connection timing info">connection start time</dfn> (default zero)
+  <dd>A <a for=/>DOMHighResTimeStamp</a>
+  <dt><dfn for="connection timing info">connection end time</dfn> (default zero)
+  <dd>A <a for=/>DOMHighResTimeStamp</a>
+  <dt><dfn for="connection timing info">secure connection start time</dfn> (default zero)
```

Comment:
It's still exposed in resource timing. @yoavweiss ?

Reply:
We need to continue exposing it, but @annevk makes a good point that currently the information it provides is not much more than "what was the scheme". Makes sense to do that composition on the Resource Timing side, and maybe we can later discuss deprecating and removing it.

Reply:
Looking deeper, in Chromium it exposes a different timer than connectStart: https://source.chromium.org/chromium/chromium/src/+/master:third_party/blink/renderer/core/timing/performance_resource_timing.cc;l=314?q=secureConnectionStart%20f:.cc&ss=chromium

I think we need to take a look at what implementations are doing and align the spec to that.

Reply:
Seems to me that it would be best to keep the "secure connection start time" value in Fetch, and add a TODO to test current implementations to see that they match the prose rather than the diagram. At least for Chromium, it seems like it's *supposed* to match the prose (i.e. `connectStart` <= `connectEnd` <= `secureConnectionStart`)

Reply:
Why would connectEnd be smaller than secureConnectionStart?

Reply:
There are 2 possible interpretations to `connectEnd`: one where it only comprises the non-TLS part of the connection in traditional TLS over TCP connection modes (i.e. `connectStart` <= `connectEnd` <= `secureConnectionStart`) , and one where it includes the secure connection establishment (i.e. `connectStart `<=  `secureConnectionStart` <= `connectEnd`). I'm not currently certain which one of those interpretations is implemented in Chromium.

Reply:
But the RT spec explicitly says that `connectEnd` should occur after the TLS handshake started... https://w3c.github.io/resource-timing/#dom-performanceresourcetiming-connectend: 

"The returned time MUST include the time interval to establish the transport connection, as well as other time intervals such as SOCKS authentication. It MUST include the time interval to complete enough of the TLS handshake to request the resource"

So I think it's actually `connectStart <= secureConnectionStart <= connectEnd` if to judge by current spec.

Reply:
You're right, the "On getting, the connectEnd.." part does clarify the above. Should we include similar prose in Fetch?

Reply:
Fixed